pkg/controller/quay_io_ci_images_distributor/mirror.go

```diff
@@ -80,3 +83,48 @@ func NewMirrorStore() MirrorStore {
 		mirrors: map[string]MirrorTask{},
 	}
 }
+
+type MirrorConsumerController struct {
+	logger            *logrus.Entry
+	quayIOImageHelper QuayIOImageHelper
+	mirrorStore       MirrorStore
+	options           OCImageMirrorOptions
+}
+
+func (c *MirrorConsumerController) Run() error {
+	for {
+		mirrors, err := c.mirrorStore.Take(10)
+		if err != nil {
+			c.logger.WithError(err).Warn("Failed to take mirrors")
+			continue
+		}
+		if len(mirrors) == 0 {
+			c.logger.Debug("Waiting for mirror tasks ...")
+			time.Sleep(3 * time.Second)
```

Comment:
Have you already considered switching to an event base approach?

Reply:
Yes. This is very good point.
The task producer is already event based.
In K8S world, typically I could generate a CRD in the producer and the consumer here watch its CR.
We do have code like this in other dptp-controllers as you might have seen.
For the simplicity, I did not go that direction for the consumer.
I want to feel how it goes in our production first and then optimize the tool for the issues there.